### PR TITLE
update deps and fix vault test suite

### LINKED ISSUE #167: Test suite fails to pull testcontainers images
### Version report

hashicorp-vault-plugin master (005818d4e7bf9e96e2d5fc7df20b7a9d8de595cb)

- What Operating System are you using (both controller, and any agents involved in the problem)?

Linux with OpenJDK 11.

### Reproduction steps

`mvn test`

### Results

Expected result:

Tests succeed.

Actual result:

```
[ERROR] com.datapipe.jenkins.vault.it.buildwrapper.SSLTest  Time elapsed: 0.917 s  <<< ERROR!
java.lang.ExceptionInInitializerError
	at java.base/jdk.internal.misc.Unsafe.ensureClassInitialized0(Native Method)
	at java.base/jdk.internal.misc.Unsafe.ensureClassInitialized(Unsafe.java:1042)
	at java.base/jdk.internal.reflect.UnsafeFieldAccessorFactory.newFieldAccessor(UnsafeFieldAccessorFactory.java:43)
	at java.base/jdk.internal.reflect.ReflectionFactory.newFieldAccessor(ReflectionFactory.java:186)
	at java.base/java.lang.reflect.Field.acquireFieldAccessor(Field.java:1105)
	at java.base/java.lang.reflect.Field.getFieldAccessor(Field.java:1086)
	at java.base/java.lang.reflect.Field.get(Field.java:418)
	at org.junit.runners.model.FrameworkField.get(FrameworkField.java:88)
	at org.junit.runners.model.TestClass.collectAnnotatedFieldValues(TestClass.java:249)
	at org.junit.runners.ParentRunner.classRules(ParentRunner.java:280)
	at org.junit.runners.ParentRunner.withClassRules(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.classBlock(ParentRunner.java:217)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:412)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:365)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:273)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:238)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:159)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:377)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:138)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:465)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:451)
Caused by: com.github.dockerjava.api.exception.NotFoundException: {"message":"No such image: quay.io/testcontainers/ryuk:0.2.3"}

	at com.github.dockerjava.okhttp.OkHttpInvocationBuilder.execute(OkHttpInvocationBuilder.java:287)
	at com.github.dockerjava.okhttp.OkHttpInvocationBuilder.execute(OkHttpInvocationBuilder.java:271)
	at com.github.dockerjava.okhttp.OkHttpInvocationBuilder.post(OkHttpInvocationBuilder.java:129)
	at com.github.dockerjava.core.exec.CreateContainerCmdExec.execute(CreateContainerCmdExec.java:33)
	at com.github.dockerjava.core.exec.CreateContainerCmdExec.execute(CreateContainerCmdExec.java:13)
	at com.github.dockerjava.core.exec.AbstrSyncDockerCmdExec.exec(AbstrSyncDockerCmdExec.java:21)
	at com.github.dockerjava.core.command.AbstrDockerCmd.exec(AbstrDockerCmd.java:35)
	at com.github.dockerjava.core.command.CreateContainerCmdImpl.exec(CreateContainerCmdImpl.java:595)
	at org.testcontainers.utility.ResourceReaper.start(ResourceReaper.java:91)
	at org.testcontainers.DockerClientFactory.client(DockerClientFactory.java:155)
	at org.testcontainers.DockerClientFactory.getActiveApiVersion(DockerClientFactory.java:323)
	at org.testcontainers.utility.TestEnvironment.dockerApiAtLeast(TestEnvironment.java:16)
	at com.datapipe.jenkins.vault.util.VaultTestUtil.hasDockerDaemon(VaultTestUtil.java:40)
	at com.datapipe.jenkins.vault.util.VaultContainer.createVaultContainer(VaultContainer.java:41)
	at com.datapipe.jenkins.vault.it.buildwrapper.SSLTest.<clinit>(SSLTest.java:42)
	... 21 more
```

You can also see this in the checks for #166.

### PR DESCRIPTION
**Pull request by jetersen**
fixes #167 

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your master branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Comment by jetersen** [comment]
@dbnicholson PTAL :)

**Review by dbnicholson — APPROVED** [review]
I ran this locally and it passed even after deleting the container image it would pull. Otherwise, I just had 2 small comments. I'd also squash this all into 1 commit, but you're the maintainer.

File: src/test/java/com/datapipe/jenkins/vault/VaultBuildWrapperTest.java [hunk 1]
```diff
@@ -18,7 +18,6 @@
 import java.util.HashMap;
 import java.util.List;
 import jenkins.tasks.SimpleBuildWrapper;
-import org.jetbrains.annotations.NotNull;
```
[the remark this answers is not in the document]
Comment (jetersen) [reply src/test/java/com/datapipe/jenkins/vault/VaultBuildWrapperTest.java:21]: Nope, license issue with javax annotation.

File: src/test/java/com/datapipe/jenkins/vault/util/VaultTestUtil.java [hunk 2]
```diff
@@ -47,25 +51,41 @@ public static VaultContainer createVaultContainer() {
...
     }
 
+    public static String getTestPath(String resource) {
+        URI uri = null;
+        try {
+            uri = VaultTestUtil.class.getResource(resource).toURI();
+        } catch (URISyntaxException e) {
+            // ignored
+        }
+        if (uri == null) return "FileNotFound"; // Should never happen.
```
[the remark this answers is not in the document]
Comment (jetersen) [reply src/test/java/com/datapipe/jenkins/vault/util/VaultTestUtil.java:73]: cause exception would propagate to the `ClassRule`
https://github.com/jenkinsci/hashicorp-vault-plugin/blob/005818d4e7bf9e96e2d5fc7df20b7a9d8de595cb/src/test/java/com/datapipe/jenkins/vault/jcasc/secrets/VaultSecretSourceTest.java#L49-L50 and that felt like bigger issue...

Not sure why this suddenly became an issue with testcontainers-java didn't care much for figuring out why it stopped accepting a test resource path, I just copied the bits that were private inside the mountable class that made it work previously 😓

File: src/test/java/com/datapipe/jenkins/vault/VaultBuildWrapperTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (dbnicholson) [reply src/test/java/com/datapipe/jenkins/vault/VaultBuildWrapperTest.java:21]: Interesting, I wasn't aware of that. Now that it's already in use here.

File: src/test/java/com/datapipe/jenkins/vault/util/VaultTestUtil.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (dbnicholson) [reply src/test/java/com/datapipe/jenkins/vault/util/VaultTestUtil.java:73]: Makes sense. It's just test code, so meh.

**Review by dbnicholson — APPROVED** [review]

File: src/test/java/com/datapipe/jenkins/vault/VaultBuildWrapperTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (jetersen) [reply src/test/java/com/datapipe/jenkins/vault/VaultBuildWrapperTest.java:21]: I know, just haven't spent time cleaning it up 😅

**Pull request merged**